src/Impl/GuzzleHttpRequester.php

```diff
@@ -44,27 +59,37 @@ public function executePost(
      * @return MessageStatus[]
      * @throws StreamxClientException
      */
-    private function handleResponse(ResponseInterface $response): array
+    private function handleIngestionResponse(ResponseInterface $response): array
     {
         $statusCode = $response->getStatusCode();
 
-        if ($statusCode == 202) {
-            return $this->parseMessageStatuses($response);
-        }
-
-        if ($statusCode == 401) {
-            throw new StreamxClientException('Authentication failed. Make sure that the given token is valid.');
+        switch ($statusCode) {
+            case 202:
+                return $this->parseMessageStatuses($response);
+            case 401:
+                throw $this->createAuthenticationException();
```

Comment:
extracted to method used also in `handleSchemaResponse`

Reply:
what about other codes like 403 forbidden or 500 ? I guess we should handle the ones that are documented by rest ingestion reference